include/ibamr/IBFESurfaceMethod.h

```diff
@@ -521,26 +566,139 @@ class IBFESurfaceMethod : public IBStrategy
+                    current_vecs[part] =
+                        dynamic_cast<libMesh::PetscVector<double>*>(systems[part]->current_local_solution.get());
+                    *current_vecs[part] = *systems[part]->solution;
+                }
+                if (d_has_half_vecs)
+                {
+                    half_vecs[part] = dynamic_cast<libMesh::PetscVector<double>*>(
+                        systems[part]->current_local_solution->clone().release());
+                    *half_vecs[part] = *systems[part]->solution;
+                }
+                if (d_has_new_vecs)
+                {
+                    new_vecs[part] = dynamic_cast<libMesh::PetscVector<double>*>(
+                        systems[part]->current_local_solution->clone().release());
+                    *new_vecs[part] = *systems[part]->solution;
+                }
+                if (d_has_IB_ghost_vecs)
+                {
+                    IB_ghost_vecs[part] = dynamic_cast<libMesh::PetscVector<double>*>(
+                        fe_data_managers[part]->buildGhostedCoordsVector(/*localize_data*/ false));
+                }
+            }
+            d_initialized = true;
+        }
+
+        void free()
+        {
+            if (d_initialized)
+            {
+                const auto n_parts = systems.size();
+                for (auto part = 0; part < n_parts; ++part)
+                {
+                    if (d_has_new_vecs)
+                    {
+                        *systems[part]->current_local_solution = *new_vecs[part];
+                    }
+                    else if (d_has_half_vecs)
+                    {
+                        *systems[part]->current_local_solution = *half_vecs[part];
+                    }
+                    else if (d_has_current_vecs)
+                    {
+                        *systems[part]->current_local_solution = *current_vecs[part];
+                    }
+                    *systems[part]->solution = *systems[part]->current_local_solution;
+                }
+
+                systems.clear();
+                current_vecs.clear();
+                if (d_has_half_vecs)
+                    std::for_each(
+                        half_vecs.begin(), half_vecs.end(), [](libMesh::PetscVector<double>* v) { delete v; });
+                half_vecs.clear();
+                if (d_has_new_vecs)
+                    std::for_each(new_vecs.begin(), new_vecs.end(), [](libMesh::PetscVector<double>* v) { delete v; });
+                new_vecs.clear();
+                IB_ghost_vecs.clear();
+                d_initialized = false;
+            }
+        }
+
+        std::vector<libMesh::System*> systems;
+        std::vector<libMesh::PetscVector<double>*> current_vecs, half_vecs, new_vecs, IB_ghost_vecs;
+
+    private:
+        const std::string d_system_name;
+        const bool d_has_current_vecs, d_has_half_vecs, d_has_new_vecs, d_has_IB_ghost_vecs;
+        bool d_initialized = false;
+    };
```

Comment:
This is really great, and makes it much more impact!

Reply:
The issue had to do with the class `LibMeshSystemData`. I have everything working by keeping all new changes except removing this class. Please let me know if you want me to push them.... If we still want to make the system of equations more compact perhaps  you and @drwells can come up with a proper way to handle them differently. Currently everything including those shared with `fe_data_manager` like `COORDINATE_SYSTEM_NAME, `VELOCITY_SYSTEM_NAME`, and `FORCE_SYSTEM_NAME` are handled through `LibMeshSystemData`, while I think that might be a good place to only keep the jump systems and not everything?